server/src/internalClusterTest/java/org/opensearch/snapshots/RemoteIndexSnapshotStatusApiIT.java

```diff
@@ -192,6 +200,110 @@ public void testStatusAPICallInProgressShallowSnapshot() throws Exception {
         createSnapshotResponseActionFuture.actionGet();
     }
 
+    public void testStatusAPICallForShallowV2Snapshot() throws Exception {
+        disableRepoConsistencyCheck("Remote store repository is being used for the test");
+        Settings pinnedTimestampSettings = Settings.builder()
+            .put(RemoteStoreSettings.CLUSTER_REMOTE_STORE_PINNED_TIMESTAMP_ENABLED.getKey(), true)
+            .build();
+        internalCluster().startClusterManagerOnlyNode(pinnedTimestampSettings);
+        internalCluster().startDataOnlyNodes(2, pinnedTimestampSettings);
+
+        final String index1 = "remote-index-1";
+        final String index2 = "remote-index-2";
+        final String index3 = "remote-index-3";
+        final String snapshotRepoName = "snapshot-repo-name";
+        final String snapshot = "snapshot";
+
+        logger.info("Create repository for shallow V2 snapshots");
+        Settings.Builder snapshotV2RepoSettings = snapshotRepoSettingsForShallowCopy().put(
+            BlobStoreRepository.SHALLOW_SNAPSHOT_V2.getKey(),
+            Boolean.TRUE
+        );
+        createRepository(snapshotRepoName, "fs", snapshotV2RepoSettings);
+
+        final Settings remoteStoreEnabledIndexSettings = getRemoteStoreBackedIndexSettings();
+        createIndex(index1, remoteStoreEnabledIndexSettings);
+        createIndex(index2, remoteStoreEnabledIndexSettings);
+        createIndex(index3, remoteStoreEnabledIndexSettings);
+        ensureGreen();
+
+        logger.info("Indexing some data");
+        for (int i = 0; i < 50; i++) {
+            index(index1, "_doc", Integer.toString(i), "foo", "bar" + i);
+            index(index2, "_doc", Integer.toString(i), "foo", "bar" + i);
+            index(index3, "_doc", Integer.toString(i), "foo", "bar" + i);
+        }
+        refresh();
+
+        SnapshotInfo snapshotInfo = createFullSnapshot(snapshotRepoName, snapshot);
+        assertTrue(snapshotInfo.getPinnedTimestamp() > 0); // to assert creation of a shallow v2 snapshot
+
+        logger.info("Set MAX_SHARDS_ALLOWED_IN_STATUS_API to a low value");
+        ClusterUpdateSettingsRequest updateSettingsRequest = new ClusterUpdateSettingsRequest();
+        updateSettingsRequest.persistentSettings(Settings.builder().put(MAX_SHARDS_ALLOWED_IN_STATUS_API.getKey(), 2));
+        assertAcked(client().admin().cluster().updateSettings(updateSettingsRequest).actionGet());
+
+        assertBusy(() -> {
+            // without index filter
+            // although no. of shards in snapshot (3) is greater than the max value allowed in a status api call, the request does not fail
```

Comment:
So, the setting is only applicable for non shallow V2 snapshots, is this understanding correct?

Reply:
Yes, added it to the description of the setting that it does not apply to shallow-V2 snapshots